pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/impl/AggregateMetricsRule.java

```diff
@@ -67,8 +67,10 @@ private boolean shouldAggregate(InputManager inputManager) {
           continue;
         }
         FunctionContext funcCtx = selectExpr.getFunction();
-        if (selectExpr.getType() != ExpressionContext.Type.FUNCTION || !funcCtx.getFunctionName()
-            .equalsIgnoreCase("SUM") || hasNonMetricArguments(funcCtx.getArguments(), metricNames)) {
+        if (selectExpr.getType() != ExpressionContext.Type.FUNCTION || (
+            !funcCtx.getFunctionName().equalsIgnoreCase("SUM") && !funcCtx.getFunctionName().equalsIgnoreCase("SUMINT")
```

Comment:
@sajjad-moradi Can you help take a look at this part? Should we also include `SUM_PRECISION` here?